pkg/test/framework/components/echo/echotest/run.go

```diff
@@ -74,11 +77,15 @@ func (t *T) Run(testFn oneToOneTest) {
 //      ingr.CallWithRetryOrFail(...)
 //    })
 func (t *T) RunFromClusters(testFn oneClusterOneTest) {
-	t.fromEachCluster(t.rootCtx, func(ctx framework.TestContext, c cluster.Cluster) {
-		t.toEachDeployment(ctx, func(ctx framework.TestContext, dstInstances echo.Instances) {
-			t.setupDst(ctx, dstInstances)
-			testFn(ctx, c, dstInstances)
-		})
+	t.toEachDeployment(t.rootCtx, func(ctx framework.TestContext, dstInstances echo.Instances) {
```

Comment:
cc @JimmyCYJ 

I inverted the loops for 2 reasons:
* we want to run the setup once per destination, the other way would require `dest * clusters` where this is only `dest` so we avoid cleanup/setup
* in single cluster we can omit the `cluster-1` part of the test name

Reply:
Also, I made sure the `SetupForDestination` method you added was supported by all our `RunX` methods.